### DISCUSSION TITLE
Setting `dead_max_jobs` in sidekiq.yml configuration

### DISCUSSION THREAD

**Discussion by calvincchen** [sidekiq/sidekiq#6428] [Q&A] [answered]
Ruby: 3.3.5
Rails: 7.1.4
Sidekiq: 7.1.4

Hey, pretty sure this is working as intended and wanted to validate my understanding.

We previously used the `sidekiq.yml` configuration file to set the `dead_max_jobs` of the morgue in production.
```
dev_and_test: &dev_and_test
  :concurrency: 5
  :queues:
    - default

development:
  <<: *dev_and_test
test:
  <<: *dev_and_test
production:
  :dead_max_jobs: 50000
```

We've been using this configuration for years, but it was recently brought to my attention that the morgue limit wasn't being set to 50000. Instead, it was falling back to the default value of 10000 and resulted in use losing a number of jobs from the DeadSet. My guess is that it has been broken since our Sidekiq 7 upgrade a few months ago and it hadn't been noticed until recently.

I reviewed the [Advanced Options docs](https://github.com/sidekiq/sidekiq/wiki/Advanced-Options) and saw this line
>The Sidekiq configuration file is a YAML file that Sidekiq server uses to configure itself

I also found [this discussion](https://github.com/sidekiq/sidekiq/discussions/5011#discussioncomment-4998997) which suggests I want to use
`Sidekiq.default_configuration[:dead_max_jobs] = 50_000` 
for my use case, so that the `dead_max_jobs` be set in both the client and the server?

Couple questions:
1. Is the change in behavior of setting `dead_max_jobs` via configuration file expected?
2. In prior versions of Sidekiq (before 7), did the configuration file apply to both the client and server? When following stacktrace, I only saw the configuration file options merged when I started the Sidekiq CLI (bin/sidekiq)
3. What are server configuration options vs. ones that apply to both the client and server?

Thanks!

**Comment by mperham** [comment, marked as answer]
1. Setting internal options like that in the config yaml is not documented and not guaranteed across major versions. It happened to work.
2. The config file only applies to server. Use Ruby if you want to configure something in all environments.
3. This is not documented as they can change. [Sidekiq::Config contains a hash of key/values](https://github.com/sidekiq/sidekiq/blob/2451d70080db95cb5f69effcbd74381cf3b3f727/lib/sidekiq/config.rb#L13) which apply to the server.

Yes, adding `Sidekiq.default_configuration[:dead_max_jobs] = 50_000` to your initializer is the right approach.
